sdk/cosmos/azure-cosmos-spark_3_2-12/src/main/scala/com/azure/cosmos/spark/ChangeFeedPartitionReader.scala

```diff
@@ -175,6 +175,10 @@ private case class ChangeFeedPartitionReader
       case ChangeFeedModes.FullFidelity | ChangeFeedModes.AllVersionsAndDeletes =>
         changeFeedItemDeserializerV1
     }
+    this.partition.endLsn.foreach { endLsn =>
```

Comment:
NIT: I am old-school - and got pretty confused about the usage of foreach on an Option[T] - I guess it is pretty elegant - but my personal opinion is that the good old if (this.partition.endLsn.isDefined) {
    ImplementationBridgeHelpers.CosmosChangeFeedRequestOptionsHelper.getCosmosChangeFeedRequestOptionsAccessor
       .setEndLSN(options, endLsn)
}

would be more readable.

Purely NIT - I will get used to it eventually, so, no strong opinion.

Reply:
Changed it.